qp_spades/tests/test_spades.py

```diff
@@ -80,46 +83,147 @@ def _generate_testing_files(self):
 
         self.aid = aid
 
-    def test_spades(self):
+    def test_spades_to_array(self):
         self._generate_testing_files()
 
-        # testing isolate/merge
+        # testing isolate/no-merge
         params = {
             'type': 'isolate', 'merging': 'no merge', 'input': self.aid,
             'threads': 5, 'memory': 200, 'k-mers': '21,33,55,77,99,127'}
-        data = {'user': 'demo@microbio.me',
-                'command': dumps(['qp-spades', '2021.05', 'spades v3.15.2']),
-                'status': 'running',
-                'parameters': dumps(params)}
-        jid = self.qclient.post('/apitest/processing_job/', data=data)['job']
         out_dir = mkdtemp()
         self._clean_up_files.append(out_dir)
+        artifact_info = self.qclient.get(f"/qiita_db/artifacts/{self.aid}/")
+        directory = {dirname(ffs) for _, fs in artifact_info['files'].items()
+                     for ffs in fs}
+        directory = directory.pop()
+        prep_info = self.qclient.get(f'/qiita_db/prep_template/{self.pid}/')
+        df = pd.read_csv(prep_info['prep-file'], sep='\t', dtype='str',
+                         na_values=[], keep_default_na=True)
+        if 'run_prefix' not in df.columns:
+            return False, None, 'Missing run_prefix column in your preparation'
+        prefix_to_name = df.set_index('run_prefix')['sample_name'].to_dict()
```

Comment:
just curious, why assert the presence of `run_prefix` but not `sample_name`?

Reply:
Also, good question. I have never seen a prep information file in Qiita without sample_name (it's the index column) so I think it's safe to assume that is always there.